shell.nix

```diff
@@ -0,0 +1,45 @@
+let 
+  pkgs = import <nixpkgs> {};
```

Comment:
This should use the pinned (niv) nixpkgs instead.

Reply:
What change do you suggest here? I'm not familiar with niv.

Reply:
Instead of `import <nixpkgs> {}`, `import (import ./nix/sources.nix).nixpkgs {}`. See https://github.com/nmattia/niv#bootstrapping-a-nix-project.

To an approximation, nix/sources.json is a (JSON representation of) a bunch of `fetchBlah` statements, and `nix/sources.nix` provides an attribute set (dict) that evaluates each of those (so `(import ./nix/sources.nix).nixpkgs` will download the thing described in the `"nixpkgs"` key).

This way, you can `niv add`/`niv update` and get the correct hashes and everything specified automatically.

Reply:
Thanks for your help @nightkr, resolved in 942b0f2dfb7548650ec5c5658d4357a3b25cff54